### PR TITLE
OPA: add HTTP send tracing tests for authorization checks, fix nil pointer error

### PR DESCRIPTION
**Pull request by mjungsbluth**
When a Rego policy was using `http.send` and also tracing was enabled, the internally used RoundTrip could be nil, raising an error during evaluation. 

This PR adds a test to replicate the behavior and fixes the issue.

### CODE REVIEW

File: filters/openpolicyagent/tracing.go [hunk 1]
```diff
@@ -49,6 +49,11 @@ func WithTracingOptManager(manager *plugins.Manager) func(*transport) {
 func (*tracingFactory) NewTransport(tr http.RoundTripper, opts opatracing.Options) http.RoundTripper {
 	log := &logging.DefaultLog{}
 
+	// If the underlying transport is nil, use the default transport
+	if tr == nil {
```
Comment (szuecs) [inline filters/openpolicyagent/tracing.go:53]: Do you  this is  a  missconfiguration, or does it happen on normal configuration?
If miss configuration I would rather fail fast, like a fatal log with a message what was missing

Comment (mjungsbluth) [reply filters/openpolicyagent/tracing.go:53]: This is sadly the normal behavior for this call path (when the http call is triggered from within a policy), for the the other path (for example comms with S3) we have a non-nil value.

**Comment by szuecs** [comment]
:+1:

**Comment by mjungsbluth** [comment]
👍
